core/src/main/scala/org/apache/spark/ui/SparkUI.scala

```diff
@@ -150,7 +151,14 @@ private[spark] object SparkUI {
       appName: String,
       basePath: String,
       startTime: Long): SparkUI = {
-    create(None, conf, listenerBus, securityManager, appName, basePath, startTime = startTime)
+    val sparkUI = create(
+      None, conf, listenerBus, securityManager, appName, basePath, startTime = startTime)
+    JsonProtocol.eventRegisters.foreach { eventRegister =>
+      val listener = eventRegister.getListener()
+      listenerBus.addListener(listener)
+      eventRegister.attachUITab(listener, sparkUI)
```

Comment:
This doesn't look right. Won't this attach, e.g. the SQL tab to every application in the history server, regardless of whether the app used SQL or not?